src/shared/sam/localLambdaRunner.ts

```diff
@@ -272,14 +345,113 @@ export async function invokeLambdaFunction(
                 })
             },
         })
-        if (attachResults.success) {
-            await showDebugConsole()
+            .then(r => {
+                if (r.success) {
+                    showDebugConsole()
+                }
+            })
+            .catch(e => {
+                getLogger().error(`Failed to debug: ${e}`)
+                ctx.chanLogger.channel.appendLine(`Failed to debug: ${e}`)
+            })
+
+        if (config.invokeTarget.target !== 'api') {
+            await tryAttach
+        } else {
+            const payload = JSON.parse(await readFile(config.eventPayloadFile, { encoding: 'utf-8' }))
+            await requestLocalApi(ctx, config.api!, config.apiPort!, payload).finally(() => {
+                stopApi(config)
+            })
+            const timer2 = createInvokeTimer(ctx.settings)
```

Comment:
Should we create a second timer? By using a second `createInvokeTimer`, we can maximally double the user-set `samcli.debug.attach.timeout.millis` value. Thinking if anything we would just keep using the first `timer` since that would represent the entire debugger lifecycle.

Reply:
The timeouts are for different components. We already have other timeouts that are not controlled by `samcli.debug.attach.timeout.millis`, it does not decide the total time waited, it only decides the time to wait for attach. Waiting for the API to standup is a different step, and could take longer that what we have accounted for the debugger.

Reply:
Should this be a different setting then? What if you're running an API that takes an inordinate amount of time to stand up?

Reply:
We may end up removing this entirely. It may not be needed since several timing-related bug fixes were done later in other areas.

Reply:
Removed it.